### PR TITLE
packer_cache: pull and tag middleware/weblogic:12.2.1.3-dev

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Cache docker images in the CI workers to speed up the build process

### CODE REVIEW

**Review by mtojek — APPROVED** [review]

**Comment by kush-elastic** [comment]
Hey @mtojek, We looked into this dockerfiles we will require access to **[container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev](http://container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev)** image as we will be using this two files, file1(https://github.com/oracle/docker-images/blob/main/OracleWebLogic/samples/12213-deploy-application/Dockerfile)  and file2(https://github.com/oracle/docker-images/blob/main/OracleWebLogic/samples/12213-domain-home-in-image/Dockerfile) as reference to have admin-server managed-server and Sample Application Deployed on WebLogic.

**Review by mdelapenya — APPROVED** [review]

**Comment by mtojek** [comment]
> We looked into this dockerfiles we will require access to [container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev](http://container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev)

You don't need access to container-registry as images will be copied to observability namespace. Unless you're asking for 12.2.1.3-dev instead of 12.2.1.4.

**Comment by kush-crest** [comment]
> > We looked into this dockerfiles we will require access to [container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev](http://container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev)
> 
> You don't need access to container-registry as images will be copied to observability namespace. Unless you're asking for 12.2.1.3-dev instead of 12.2.1.4.

Yes, we would explicitly require [container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev](http://container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev) image.

File: .ci/packer_cache.sh [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#!/usr/bin/env bash
+
+# shellcheck disable=SC1091
+source /usr/local/bin/bash_standard_lib.sh
+
+if [ -x "$(command -v docker)" ]; then
+  IMAGE=docker.elastic.co/observability-ci/weblogic:12.2.1.4
```
Comment (v1v) [inline .ci/packer_cache.sh:7]:
```suggestion
  IMAGE=docker.elastic.co/observability-ci/weblogic:12.2.1.3-dev
```

File: .ci/packer_cache.sh [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  (retry 2 docker pull "${IMAGE}") || echo "Error pulling ${IMAGE} Docker image, we continue"
+  docker tag "${IMAGE}" container-registry.oracle.com/middleware/weblogic:12.2.1.4
```
Comment (v1v) [inline .ci/packer_cache.sh:9]:
```suggestion
  docker tag "${IMAGE}" container-registry.oracle.com/middleware/weblogic:12.2.1.3-dev
```

**Review by mtojek — APPROVED** [review]

**Comment by kush-elastic** [comment]
Hey @v1v, @mtojek We tried pulling the docker image `docker pull docker.elastic.co/observability-ci/weblogic:12.2.1.3-dev`  in local as well as CI its giving `docker login error` can you help us with this we also tried login with elastic account in local but its giving same error  

```
[2022-05-11T05:17:15.735Z] Building wlsadmin

[2022-05-11T05:17:15.998Z] Step 1/29 : FROM docker.elastic.co/observability-ci/weblogic:12.2.1.3-dev

[2022-05-11T05:17:15.998Z] Service 'wlsadmin' failed to build: pull access denied for docker.elastic.co/observability-ci/weblogic, repository does not exist or may require 'docker login': denied: requested access to the resource is denied

[2022-05-11T05:17:15.998Z] Error: error running package system tests: could not complete test run: could not setup service: could not boot up service using Docker Compose: running Docker Compose up command failed: exit status 1

script returned exit code 1
```

Error reference in CI: https://beats-ci.elastic.co/blue/organizations/jenkins/Ingest-manager%2Fintegrations/detail/PR-3121/11/pipeline/

**Comment by v1v** [comment]
`packer_cache.sh` takes one or two days to be run, unfortunately, the packer_cache automation uses master or main, but in the integrations project both branches were in place, therefore it used `master` which it's not up-to-date

image

I just deleted `master` branch and therefore the packer cache images might be available sometime today,

image

If further details are needed, then please contact the CI system teams are they are the CI system owners and more knowledgeable

**Comment by kush-elastic** [comment]
> `packer_cache.sh` takes one or two days to be run, unfortunately, the packer_cache automation uses master or main, but in the integrations project both branches were in place, therefore it used `master` which it's not up-to-date
> 
> image
> 
> I just deleted `master` branch and therefore the packer cache images might be available sometime today,
> 
> image
> 
> If further details are needed, then please contact the CI system teams since they are the CI system owners and more knowledgeable

Okay, Thanks 👍

**Pull request merged**